app/src/main/java/org/mozilla/fenix/settings/logins/EditLoginFragment.kt

```diff
@@ -0,0 +1,207 @@
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.components
+
+/**
+ * Displays the editable saved login information for a single website.
+ */
+class EditLoginFragment : Fragment(R.layout.fragment_edit_login) {
+
+    private val args by navArgs<EditLoginFragmentArgs>()
+    private lateinit var savedLoginsStore: LoginsFragmentStore
+    fun String.toEditable(): Editable = Editable.Factory.getInstance().newEditable(this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+
+        savedLoginsStore = StoreProvider.get(this) {
+            LoginsFragmentStore(
+                LoginsListState(
+                    isLoading = true,
+                    loginList = listOf(),
+                    filteredItems = listOf()
+                )
+            )
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // ensure hostname isn't editable
+        hostnameText.text = args.savedLoginItem.origin.toEditable()
+        hostnameText.isClickable = false
+        hostnameText.isFocusable = false
+
+        usernameText.text = args.savedLoginItem.username.toEditable()
+        passwordText.text = args.savedLoginItem.password!!.toEditable()
+
+        // TODO: extend PasswordTransformationMethod() to change bullets to asterisks
+        passwordText.inputType =
+            InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+
+        setUpClickListeners()
+    }
+
+    private fun setUpClickListeners() {
+        clearUsernameTextButton.setOnClickListener {
+            usernameText.text?.clear()
+            usernameText.isCursorVisible = true
+            usernameText.hasFocus()
+            inputLayoutUsername.hasFocus()
+        }
+        clearPasswordTextButton.setOnClickListener {
+            passwordText.text?.clear()
+            passwordText.isCursorVisible = true
+            passwordText.hasFocus()
+            inputLayoutPassword.hasFocus()
+        }
+        revealPasswordButton.setOnClickListener {
+            togglePasswordReveal()
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.login_save, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean = when (item.itemId) {
+        R.id.save_login_button -> {
+            closeKeyboard()
+            try {
+                if (!passwordText.text.isNullOrBlank()) {
+                    attemptSaveAndExit()
+                } else {
+                    view?.let {
+                        FenixSnackbar.make(
+                            view = it,
+                            duration = Snackbar.LENGTH_SHORT,
+                            isDisplayedWithBrowserToolbar = false
+                        ).setText("Password cannot be blank.").show()
```

Comment:
I think this is supposed to show an error state that says "Password required"? We should probably also update that error state as the user is typing and not just when they submit. If we still want to show the snackbar, it also needs to be pulled into strings?